### ISSUE TITLE
obsolete Legality in Standard format

### ISSUE THREAD

**Issue by NikolayXHD** [mtgjson/mtgjson#479]
Hi! Card legalities for `Standard` format are obsolete in `AllSets-x.json` file.
`XLN` set cards are not yet legal while `SOI` and `EMN` are still legal.

**Comment by bakert** [comment]
The XLN issue is because C.js has it as IXN not XLN. XLN is correct.

**Comment by bakert** [comment]
I rolled this into #486 but I'm not sure it's very mergeable. I made the change as a separate commit so a more experienced mtgjson-er should feel free to cherry pick it and do the right thing.

**Comment by fenhl** [comment]
Has this been fixed in #486?

**Comment by NikolayXHD** [comment]
Yes it was. Could someone please update mtgjson.com website? I am asking because to me as a Windows user it is rather tricky to build the allsets-x.json file myself.

There is a couple of problems that make building the file harder than fixing legality problem on side of application using allsets-x.json.

In case someone is curious what are the problems:

I've managed to `npm install` all the modules after having installed `nodejs 8.9.1` and  `python 2.7.14` and but
1. The CON.* files are banned in windows so some workaround is needed.
2. When trying to `npm build XLN` I get
```
PS F:\repo\git\mtgjson\build> node .\buildSet.js XLN                            
F:\repo\git\mtgjson\node_modules\bindings\bindings.js:91                        
  throw err                                                                     
  ^                                                                             
                                                                                
Error: Could not locate the bindings file. Tried:                               
 → F:\repo\git\mtgjson\node_modules\mhash\build\mhash.node                      
 → F:\repo\git\mtgjson\node_modules\mhash\build\Debug\mhash.node                
 → F:\repo\git\mtgjson\node_modules\mhash\build\Release\mhash.node              
 → F:\repo\git\mtgjson\node_modules\mhash\out\Debug\mhash.node                  
 → F:\repo\git\mtgjson\node_modules\mhash\Debug\mhash.node                      
 → F:\repo\git\mtgjson\node_modules\mhash\out\Release\mhash.node                
 → F:\repo\git\mtgjson\node_modules\mhash\Release\mhash.node                    
 → F:\repo\git\mtgjson\node_modules\mhash\build\default\mhash.node              
 → F:\repo\git\mtgjson\node_modules\mhash\compiled\8.9.1\win32\x64\mhash.node   
    at bindings (F:\repo\git\mtgjson\node_modules\bindings\bindings.js:88:9)    
    at Object.<anonymous> (F:\repo\git\mtgjson\node_modules\mhash\index.js:1:99)
    at Module._compile (module.js:635:30)                                       
    at Object.Module._extensions..js (module.js:646:10)                         
    at Module.load (module.js:554:32)                                           
    at tryModuleLoad (module.js:497:12)                                         
    at Function.Module._load (module.js:489:3)                                  
    at Module.require (module.js:579:17)                                        
    at require (internal/module.js:11:18)                                       
    at Object.<anonymous> (F:\repo\git\mtgjson\shared\shared.js:5:12)           
PS F:\repo\git\mtgjson\build>                                                   
```

**Comment by tooomm** [comment]
@NikolayXHD did you check the release notes? https://mtgjson.com/changelog.html
>3.11.8
>Nov 7, 2017
>
>    DDT and IMA added
    SOI, EMN Standard legality updated
    **IXN corrected to XLN in Standard legality**

No idea why @bakert put 7th nov as date in the changelog, while his pr is from 9th nov. https://github.com/mtgjson/mtgjson/pull/486

**Comment by NikolayXHD** [comment]
Thanks for hint about release notes! I did not check them, however I did check the downloaded `allsets-x.json` file visually.  It still has icncorrect `standard` legality for `xln`:
```
"XLN": {
    "name": "Ixalan",
    "code": "XLN",
...
    "cards": [
      {
        ...

        "id": "ccbc0aca09ec754dd5dc97081315dfdff1748132",
        "imageName": "adanto vanguard",
        ...

        "legalities": [
          {
            "format": "Commander",
            "legality": "Legal"
          },
          {
            "format": "Ixalan Block",
            "legality": "Legal"
          },
          {
            "format": "Legacy",
            "legality": "Legal"
          },
          {
            "format": "Modern",
            "legality": "Legal"
          },
          {
            "format": "Vintage",
            "legality": "Legal"
          }
        ],
        
        ...
        
        "name": "Adanto Vanguard",
        
        ...
```

**Comment by bakert** [comment]
I see the same thing locally for obvious reasons. I guess whatever I did to generate the files didn't regenerate the Standard legality.

`howto.txt` says:

     Sets with "recalculateStandard" parameter, will have its Standard legalities recalculated by mtgjson (using buildSet). This is currently set to ALL sets.

So it should have worked when I regenerated XLN.

If I run:

    node util/updateLegalitiesFromSet.js XLN

I get quite a few updates for basic lands and the checklands in M10 and similar sets. But json/XLN.json doesn't change at all.

What do I need to do to make XLN.json get entries for Standard?

`howto.txt` mentions a `recalculateStandard` param but scanning the code that seems like it might be obsolete?

If someone gives me a hint I'll try and drag this over the line :)

**Comment by NikolayXHD** [comment]
Maybe you did not clear legality cache?

Gatherer New Set Instructions
---------------------------------
...
So set yourself a reminder for a day after the official release date of the set to do:
```
node util/clearCacheForSet.js legalities <set codes that rotated out AND the new set code>
node build/buildSet.js <set codes that rotated out AND the new set code>
node util/updateLegalitiesFromSet.js <set codes that rotated out AND the new set code>
```
...

**Comment by NikolayXHD** [comment]
Fixed in 3.12.0

**Comment by NikolayXHD** [comment]
Thanks for help!
